### ISSUE TITLE
README Example Fails

### ISSUE THREAD

**Issue by seanlaw** [louisabraham/pydivsufsort#22]
I don't know if this has to do with the way that the package has been laid out but, at first, I tried to install the package locally via:

```
python -m pip install .
```

This is command pretty typical and things seemed to install but some of the tests failed. Then I noticed that the README actually says

```
python -m pip install pydivsufsort
```

Of course, this also installed without any error but now the tests pass. 🤷‍♂ 

And then, this usage command only works if you add `import numpy as np`. I recommend that we add this so that users can quickly and easily copy/paste

```
from pydivsufsort import divsufsort, kasai

print(divsufsort(b"abcd"))

inp = np.array([1, 4, 2, 4, 5, 2, 2])
sa = divsufsort(inp)
print(kasai(inp, sa))
```

It might even be better to use a [Wikipedia example](https://en.wikipedia.org/wiki/LCP_array). Minimally, I think the variable names should be more verbose:

```
from pydivsufsort import divsufsort, kasai
import numpy as np

print(divsufsort(b"abcd"))

inp = np.array([1, 4, 2, 4, 5, 2, 2])
suffix_array = divsufsort(inp)
lcp_array = kasai(inp, sa)

print(suffix_array, lcp_array)
```

**Comment by seanlaw** [comment]
@louisabraham Btw, I hope that you and your loved ones are staying safe!

**Comment by louisabraham** [comment]
> some of the tests failed
what do you mean?

Feel free to modify the examples. In the latest version (0.0.3), strings can be used as inputs.

**Comment by seanlaw** [comment]
This is what I get when I use `python -m pip install .`

```
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 3.7.1, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /Users/law978/Git/pydivsufsort.git
plugins: cov-2.8.1, flask-1.0.0
collected 5 items

tests/test_correct.py ....F                                                                                                                                                                          [100%]

================================================================================================= FAILURES =================================================================================================
________________________________________________________________________________________________ test_kasai ________________________________________________________________________________________________

    def test_kasai():
        inp = np.array(list(b"banana"), dtype="uint8")
        out = np.array([1, 3, 0, 0, 2, 0])
>       assert (kasai(inp) == out).all()

tests/test_correct.py:76:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pydivsufsort/stringalg.pyx:88: in pydivsufsort.stringalg.kasai
    return _kasai(s, sa)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   def _kasai(string_t[::1] s not None, sa_t[::1] sa not None):
E   TypeError: Function call with ambiguous argument types

pydivsufsort/stringalg.pyx:27: TypeError
========================================================================================= short test summary info ==========================================================================================
FAILED tests/test_correct.py::test_kasai - TypeError: Function call with ambiguous argument types
======================================================================================= 1 failed, 4 passed in 1.38s ========================================================================================
```

**Comment by louisabraham** [comment]
Please pull the latest commit. Sometimes I push commits with bugs.

**Comment by seanlaw** [comment]
> Feel free to modify the examples. In the latest version (0.0.3), strings can be used as inputs.

Will do!

**Comment by seanlaw** [comment]
Not sure why the builds are failing but it doesn't seem to be related to changing the README

**Comment by louisabraham** [comment]
About the current failures on Travis, we just have to wait for manylinux to fix them pypa/manylinux#512
